.cirrus.yml

```diff
@@ -0,0 +1,49 @@
+task:
+  trigger_type: manual
+  name: E2E/Integration Tests
+  timeout_in: 45
+  container:
+    image: ubuntu:21.04
+    kvm: true # required for E2E/Integrations tests
+    cpu: 4
+    memory: 8G
+  env:
+    DEBIAN_FRONTEND: noninteractive
+    CI: true
+    FORCE_COLOR: 1 # force terminal color
+    CIRRUS_CLONE_DEPTH: 0 # Required to get app version
+    PLAYWRIGHT_SKIP_BROWSER_DOWNLOAD: 1 # disable auto-browser download
+    KEYRING: /usr/share/keyrings/nodesource.gpg
+    NODE_VERSION: node_14.x # set node version e.g: "node_16.x" for Node 16 LTS
+    DISTRO: hirsute # release codename for Ubuntu 21.04 - see "lsb_release -s -c"
+  install_deps_script: |
+    apt-get update
+    apt-get install -y --no-install-recommends ca-certificates gcc g++ curl git golang openssh-client make netcat sudo vim gpg \
+                    libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
+    curl -sLf -o /dev/null "https://deb.nodesource.com/"$NODE_VERSION"/dists/"$DISTRO"/Release"
+    curl -s https://deb.nodesource.com/gpgkey/nodesource.gpg.key | gpg --dearmor | tee "$KEYRING" >/dev/null
+    echo "deb [signed-by="$KEYRING"] https://deb.nodesource.com/"$NODE_VERSION" "$DISTRO" main" > /etc/apt/sources.list.d/nodesource.list
+    echo "deb-src [signed-by="$KEYRING"] https://deb.nodesource.com/"$NODE_VERSION" "$DISTRO" main" >> /etc/apt/sources.list.d/nodesource.list
+    apt-get update && apt-get install -y  nodejs && node -v && npm -v
+
+  # Tests should run as non-root user - setting up a test user
+  prepate_user_script:
+    - groupadd -g $(stat -c '%g' /dev/kvm) kvm
```

Comment:
The group owning `/dev/kvm` doesn't already exist? That's unusual (and probably deserves a comment).

Reply:
I thought it exists already, but it does not. See: https://cirrus-ci.com/task/6448951910989824

Reply:
Yep, that's fine — comment please.

Reply:
nvm,
Just got a error related to the kvm group... it already exists.
I've seen some execution that I was creating the group at every execution, but after adding the cache, it does not require to create the kvm group. 
So, not sure why some executions already has kvm group and others does not. 
A possible solution is passing `--force` as an argument in order to self validate it the group already exists or not.
`groupadd -f -g $(stat -c '%g' /dev/kvm) kvm`

```bash
-f exit successfully if the group already exists, and cancel -g if the GID is already used
```
If the group exist, there's no reason for failing the entire execution.

Reply:
@mook-as 
I've found the issue, weird one, but haven't found the root cause though.
For some reason, I cannot create/add a group after installing the deps/tools. It wasn't happening before changing the nodejs installation method, but I got it fixed setting the group and user before those installs. Hope it fix for good.
Ref: https://cirrus-ci.com/task/5252227798401024

Reply:
Given that the error is `groupadd: group 'kvm' already exists`, I would have suspected that something in the step in the middle caused creation of that group.  But [the logs](https://cirrus-ci.com/task/5719850147381248?logs=install_deps#L1868) only show `crontab` and `lpadmin` being created… I wonder if some inconsistency meant _any_ group being created (possibly only as an install hook) caused the `kvm` group to be created?

Reply:
My guess is:
Someone are taking the groupId (108) from kvm before we create the group. Creating the group before other process takes the KVM groupId did the trick. It's working though :)